### ISSUE TITLE
spell reflection and spellsteal

### ISSUE THREAD

**Issue by Summoning01** [ccshiro/cc-buglist#2238]
i'm putting this here in case someone knows more about this topic, since all of comments below seem to have a different idea of how it works. currently on the test realm nothing happens when a mage spellsteals into reflect (not even diminished buff duration). a video proof would be perfect...

By Zeshar (2,763 – 1·13·35) on 2008/05/20 (Patch 2.4.2)
_Be warned: If this spell is reflected by a warrior, you will lose one of your own buffs.
Fortunately, the warrior does not gain the buff you lost. Though I doubt you'll lose any sleep over a warrior buffed with Arcane Intellect._ [wowhead:spellsteal](http://www.wowhead.com/spell=30449/spellsteal#comments:0)

By Zeshar (2,763 – 1·13·35) on 2008/05/20 (Patch 2.4.2)
_If you happen to reflect a mages' spellsteal ability, the mage will lose one of their buffs.
This always puzzled me as in one of those paradoxical situations where "What would happen if you spell reflected a spell steal?"
Now you don't have to level a Mage and Warrior to 70 on separate accounts to find out._
molly2k on 2008/09/24 (Patch 2.4.3)
_I actually asked my friends to try this out :P Spellreflecting spellsteal actually steals one of the mage's buffs and gives it to the warrior (The warrior later joked about doing this to get molten armor, putting on caster-gear and shout "I´m a mage", not sure if he did though :P) As a rule of thumb when thinking about how spellreflect works, imagine that the warrior is casting the spell, but with the mage's stats, buffs and specc._
By Battleguard (6,179 – 1·10·73) on 2008/03/08 (Patch 2.3.3)
_"A few" things worth noting:_
- _The reflected spell will work as if the original caster cast it on him/herself. This means that its damage (if applicable) will not be modified by the 10% outgoing damage reduction from Defensive Stance (or any other damage modifiers affecting you)._
- _For the exact same reason, spells which drain/steal health will steal and return health to the caster instead of you._
- _Certain channeled spells such as Arcane Missiles will only have the first tick reflected._
- _Single target spells which have a missile-like animation (i.e. Fireball) will fly to you, then back to the target, while channeled spells (except for the ones mentioned in the preceding note) will only appear at the hands of the caster._
- _Certain spells (usually monster abilities) cannot be reflected (such as Howling Dust) - they will hit you and Spell Reflection will not fade; other abilities may have their effects negated, taking away Spell Reflection, but without damaging/debuffing the caster._
- _Not only spells can be reflected. Works with most curses, some diseases and a few poison effects as well._
- _It has to be in effect when the spell is cast AND when it hits you in order to reflect it (most noticeable with Incinerate)._
- _The reflected ability can be resisted (uses spell hit chance of the caster). It is also possible that the caster will be immune to the effect (like with most elementals)._
- _Elementals' magic-based melee attack damage will not be reflected._
- _Sometimes it will reflect more than one spell. It usually happens when the spells hit at the exact same time._ [wowhead:spell reflection](http://www.wowhead.com/spell=23920/spell-reflection#comments)

Oct 03 2009 at 3:12 AM ArtemisEnteri
_"You can reflect spellsteal, it steals a random buff from your opponent."...
"edit: I think it might just dispel a buff, rather than steal it actually. Though the idea of a prot warrior running around with ice barrier up is glorious._ [allakhazam](http://wow.allakhazam.com/forum.html?forum=243&mid=1249840925181138908)

mmmpancakes
(Topic Creator)7 years ago#6
_"Actually I know what happens when you spell steal and it gets reflected back at you.
You steal your own buff and the buff you steal will be 2 minutes only. Basically you gimp your buff."_ [gamefaqs](http://www.gamefaqs.com/boards/534914-world-of-warcraft/44951001)

**Comment by Schaka** [comment]
Yes, any form of dispel effect that can be reflected will actually remove the buffs from the caster.
I used to spam dispel into random reflects on retail sometimes and they always got my fucking shield off. That's how I remember.

**Comment by Summoning01** [comment]
found a few more possible issues:
- reflected shadowburn grants a soul shard if the lock dies.; [wowwiki](http://wowwiki.wikia.com/wiki/Reflect?oldid=1580129) says no:
  _"Neither Shadowburn's debuff nor Drain Soul will take effect when reflected."_
  atm,reflected drain soul does not grant a shard nor does it damage the caster (on the other hand, reflected drain life does damage)
- reflected seed of corruption explodes at caster's enemies instead of allies; [wowwiki](http://wowwiki.wikia.com/wiki/Reflect?oldid=1580129) says:
  _"Seed of Corruption does not deal damage to the caster's enemies but to the target's allies. This means that a reflected SoC will deal damage to its caster's pet and allies."_
  [By BloodScribe on 2010/05/10 at 2:36 PM (Patch 3.3.3)](http://www.wowhead.com/forums&topic=151296.3)
  _"when you used to be able to reflect seed of corruption, i don't know if you still can as I've never done it a second time and never really heard from anyone else doing it. it was during a game of AV and we were fighting on the bridge, and large group was clustered togther, and few of my pvp friends were aoe away at them and I was targeting a mage that was casting pyro on me when **i hit spell reflect**, next thing I see is it wear off and **watched as a lock in the middle of the group was ticking away and boom 9 killing blows appeared on my screen** :S had to look at my combat log to confirm what i just saw lol"_ https://github.com/ccshiro/cc-buglist/issues/1920
